src/app/core/user.service.ts

```diff
@@ -42,6 +48,24 @@ export class UserService {
       : Promise.resolve(false);
   }
 
+  searchUsers(query: {}) {
+    return this.http
+      .get(
+        environment.apiURL + '/users/search?text=' + query,
+        {
+          withCredentials: true
+        }
+      )
+      .toPromise()
+      .then(val => {
+        // val contains a list of users
+        const json = JSON.parse(val['_body']);
```

Comment:
Should be able to use `val.json()`, so you don't have to manually parse out the response body

Reply:
Also, use `User.instantiate()` to convert elements in the payload to User entities